sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/DocumentQueryExecutionContextBase.java

```diff
@@ -218,6 +219,18 @@ public CosmosQueryRequestOptions getFeedOptions(String continuationToken, Intege
             requestHeaders.put(HttpConstants.HttpHeaders.POPULATE_QUERY_METRICS, String.valueOf(cosmosQueryRequestOptions.isQueryMetricsEnabled()));
         }
 
+        if (cosmosQueryRequestOptions.getDedicatedGatewayRequestOptions() != null &&
```

Comment:
Is there any way to avoid the code duplication in RxDocumentClientImpl.java and here?

Reply:
Done.